erc_20.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.0.0/contracts/token/ERC20/ERC20.sol";
+
+contract SolicyCoinERC20 is ERC20 {
+    mapping(address => uint256) balances;
+    mapping(address => bool) blackList;
+    uint8 _decimal;
+    uint8 _fee;
+    address public owner;
+    uint256 totalSupply_;
+
+    string public constant name = "Solicy";
+    string public constant symbol = "Sol";
+    uint8 public constant decimals = 18;
+
+    constructor(uint256 total, uint8 __decimal, uint8 __fee) ERC20("Solicy", "SOL"){
+        owner = msg.sender;
+        _decimal = __decimal;
+        _fee = __fee;
+        totalSupply_ = total;
+        balances[msg.sender] = totalSupply_;
+        _mint(msg.sender, 10 ** 10);
+    }
+
+    modifier onlyOwner {
+        require(msg.sender == owner);
+        _;
+    }
+
+    modifier tockenOwner {
+        require(balances[msg.sender] > 0);
+        _;
+    }
+
+    function addToBlackList(address user) external {
+        blackList[user] = true;
+    }
+
+    function delFromBlackList(address user) external {
+        delete blackList[user];
+    }
+
+    function burn(address account, uint256 amount) external virtual {
```

Comment:
Here you should use modifier only token owner, in this case anyone can burn other person tokens